### PR TITLE
Alien Changes

### PR DESCRIPTION
**Pull request by Seggan**
## Changes
I added 2 new mobs to Titan: the leech and the skywhale. Skywhales are giant passive phantoms that don't burn, leeches have a chance to steal an item from your inventory when attacking. Also changed up Titan generation a little, and changed ice types of Europa and Enceladus. Mobs: moved spawning to a ticker and expanded the API.

## Related Issues
N/A

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break world generation, mob spawning, and the like.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

**Comment by ProfElements** [comment]
👍 LGTM

File: src/main/java/io/github/addoncommunity/galactifun/base/aliens/Leech.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                    damager.getAttribute(Attribute.GENERIC_MAX_HEALTH).getBaseValue(),
+                    damager.getHealth() + 2)
+                );
+            }
+        }
+    }
+
+    @Override
+    public void onDeath(@Nonnull EntityDeathEvent e) {
+        e.getDrops().clear();
```
Comment (Mooy1) [inline src/main/java/io/github/addoncommunity/galactifun/base/aliens/Leech.java:59]: store eaten items in persistent data so they aren't lost across restarts

File: src/main/java/io/github/addoncommunity/galactifun/core/listener/CelestialListener.java [hunk 2]
```diff
@@ -59,11 +60,11 @@ public void onPlayerTeleport(@Nonnull PlayerTeleportEvent e) {
     
     @EventHandler
     public void onCreatureSpawn(@Nonnull CreatureSpawnEvent e) {
+        // don't want to prevent mobs from spawning from spawners
         if (e.getSpawnReason() == CreatureSpawnEvent.SpawnReason.NATURAL) {
-            AlienWorld world = AlienWorld.getByWorld(e.getEntity().getWorld());
-            if (world != null) {
-                e.setCancelled(true);
-                world.onMobSpawn(e);
+            CelestialWorld world = CelestialWorld.getByWorld(e.getEntity().getWorld());
```
Comment (Mooy1) [inline src/main/java/io/github/addoncommunity/galactifun/core/listener/CelestialListener.java:65]: why change to celestial world

**Review by Mooy1 — APPROVED** [review]

**Review by Mooy1 — CHANGES_REQUESTED** [review]
accidently approved in the other one

File: src/main/java/io/github/addoncommunity/galactifun/core/listener/CelestialListener.java (hunk 2 above)
Comment (Seggan) [reply src/main/java/io/github/addoncommunity/galactifun/core/listener/CelestialListener.java:65]: oops

**Pull request merged**